pennylane/devices/default_mixed.py

```diff
@@ -449,6 +452,62 @@ def _apply_density_matrix(self, state, device_wires):
             )
             self._state = self._asarray(rho, dtype=self.C_DTYPE)
 
+    def _generate_projector_diagonal(self, wire):
+        """Computes the diagonal elements of the projector onto the subspace
+        generated by the given wire being in state |0>. The projector is given
+        by Id ⊗ |0><0| ⊗ Id, where |0><0| acts on `wire` and the left and
+        right identity operators on the remaining wires.
+        
+        **Example**
+        For 3 wires, the possible diagonals generated are:
+        wire = 0  ->  [1, 1, 1, 1, 0, 0, 0, 0]
+        wire = 1  ->  [1, 1, 0, 0, 1, 1, 0, 0]
+        wire = 2  ->  [1, 0, 1, 0, 1, 0, 1, 0]
+        """
+
+        n_wires_right = self.num_wires-wire-1
+        n_wires_left = wire
+        id_dim_right = 2**n_wires_right
+        id_dim_left = 2**n_wires_left
+
+        diag = [1]*(id_dim_right) + [0]*(id_dim_right)
+        diag = diag*(id_dim_left)
+
+        return diag
+
+    def _apply_measurement(self, meas_op):
+        zero_projector = self._generate_projector_diagonal(meas_op.wire)
+        one_projector = [int(not b) for b in zero_projector]
+
+        pre_op_state = self._state
+        self._apply_diagonal_unitary(zero_projector, self.wires)
+        zero_state = self.state
+
+        self._state = pre_op_state
+        self._apply_diagonal_unitary(one_projector, self.wires)
+        one_state = self.state
+
+        rho = zero_state + one_state
```

Comment:
Is this normalized ?

Reply:
In other words, if the post measurement state for one outcome is:
![equation](http://www.sciweavers.org/tex2img.php?eq=%5Crho_m%20%3D%20%5Cfrac%7BP_m%20%5Crho%20P_m%7D%7Bp_m%7D&bc=White&fc=Black&im=jpg&fs=12&ff=arev&edit=0)
and the combined post-measurement state is:
![equation](https://bit.ly/3Lpc0qU)
Then:
![equation](http://www.sciweavers.org/tex2img.php?eq=%5Crho%27%20%3D%20%5Csum_m%20P_m%20%5Crho%20P_m&bc=White&fc=Black&im=jpg&fs=12&ff=arev&edit=0)